{"schemaVersion":{"major":0,"minor":3,"patch":0},"kind":"symbol","metadata":{"symbolKind":"method","fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"getRetryWaitTime"},{"text":"(","kind":"text"},{"text":"error","kind":"externalParam"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP","text":"Error"},{"text":", ","kind":"text"},{"kind":"externalParam","text":"attempt"},{"text":": ","kind":"text"},{"text":"Int","kind":"typeIdentifier","preciseIdentifier":"s:Si"},{"text":") -> ","kind":"text"},{"text":"RetryStatus","preciseIdentifier":"s:8SotoCore11RetryStatusO","kind":"typeIdentifier"},{"text":"?","kind":"text"}],"modules":[{"name":"SotoCore"}],"roleHeading":"Instance Method","required":true,"title":"getRetryWaitTime(error:attempt:)","role":"symbol","externalID":"s:8SotoCore11RetryPolicyP03getC8WaitTime5error7attemptAA0C6StatusOSgs5Error_p_SitF"},"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)"},"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy"]]},"primaryContentSections":[{"kind":"declarations","declarations":[{"languages":["swift"],"tokens":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"kind":"identifier","text":"getRetryWaitTime"},{"kind":"text","text":"("},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"Error","kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP"},{"text":", ","kind":"text"},{"text":"attempt","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"Int","kind":"typeIdentifier","preciseIdentifier":"s:Si"},{"text":") -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore11RetryStatusO","text":"RetryStatus","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus"},{"kind":"text","text":"?"}],"platforms":["macOS"]}]},{"kind":"parameters","parameters":[{"content":[{"inlineContent":[{"text":"Error returned by HTTP client","type":"text"}],"type":"paragraph"}],"name":"error"},{"name":"attempt","content":[{"inlineContent":[{"text":"Retry attempt number","type":"text"}],"type":"paragraph"}]}]}],"sections":[],"abstract":[{"type":"text","text":"Returns whether we should retry and how long we should wait before retrying"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)"]}],"references":{"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy","abstract":[{"type":"text","text":"Protocol for Retry strategy. Has function returning amount of time before the next retry after an HTTP error"}],"url":"\/documentation\/sotocore\/retrypolicy","role":"symbol","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"RetryPolicy"}],"title":"RetryPolicy","navigatorTitle":[{"kind":"identifier","text":"RetryPolicy"}],"kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore":{"type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","kind":"symbol","url":"\/documentation\/sotocore","abstract":[{"type":"text","text":"The core framework for Soto, a Swift SDK for AWS"}],"role":"collection","title":"SotoCore"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy/getRetryWaitTime(error:attempt:)":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)","abstract":[{"text":"Returns whether we should retry and how long we should wait before retrying","type":"text"}],"url":"\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)","required":true,"type":"topic","role":"symbol","title":"getRetryWaitTime(error:attempt:)","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"getRetryWaitTime","kind":"identifier"},{"text":"(","kind":"text"},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP","text":"Error"},{"kind":"text","text":", "},{"text":"attempt","kind":"externalParam"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:Si","text":"Int"},{"kind":"text","text":") -> "},{"text":"RetryStatus","preciseIdentifier":"s:8SotoCore11RetryStatusO","kind":"typeIdentifier"},{"text":"?","kind":"text"}],"kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryStatus":{"type":"topic","abstract":[{"type":"text","text":"Return value for "},{"type":"codeVoice","code":"RetryPolicy.getRetryWaitTime"},{"text":". Either retry after time amount or donâ€™t retry","type":"text"}],"url":"\/documentation\/sotocore\/retrystatus","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus","title":"RetryStatus","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"RetryStatus","kind":"identifier"}],"navigatorTitle":[{"kind":"identifier","text":"RetryStatus"}],"kind":"symbol","role":"symbol"}}}